src/store/actionCreators/fenceConnections.ts

```diff
@@ -0,0 +1,83 @@
+import isEmpty from 'lodash/isEmpty';
+import { ThunkAction } from 'redux-thunk';
+
+import { getFenceUser } from 'services/fence';
+
+import { Api } from '../apiTypes';
+import {
+  Connection,
+  FenceConnections,
+  FenceConnectionsActions,
+  FenceConnectionsActionTypes,
+} from '../fenceConnectionsTypes';
+import { FenceName } from '../fenceTypes';
+import { RootState } from '../rootState';
+import { selectFenceConnections } from '../selectors/fenceConnections';
+
+const { entries, keys } = Object;
+
+const addFenceConnection = (
+  fenceName: FenceName,
+  connection: Connection,
+): FenceConnectionsActionTypes => ({
+  type: FenceConnectionsActions.addFenceConnection,
+  fenceName,
+  connection,
+});
+
+const toggleIsFetchingAllFenceConnections = (isLoading: boolean): FenceConnectionsActionTypes => ({
+  type: FenceConnectionsActions.toggleIsFetchingAllFenceConnections,
+  isLoading,
+});
+
+const shouldFetchConnections = (fenceName: FenceName, state: RootState): boolean =>
+  isEmpty(selectFenceConnections(state)[fenceName]);
+
+const fetchFencesConnections = (
+  api: Api,
+  fenceName: FenceName,
+): ThunkAction<void, RootState, null, FenceConnectionsActionTypes> => async (dispatch) => {
+  try {
+    const connection = await getFenceUser(api, fenceName);
+    dispatch(addFenceConnection(fenceName, connection));
+  } catch (error) {
+    console.error(`Error fetching fence connection for '${fenceName}': ${error}`);
```

Comment:
In current code (freactal), there are no error management. However, I prepared the ground for the future if we need to manage it.